src/Symfony/Component/HttpClient/Response/PluggableResponse.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Response;
+
+use Symfony\Component\HttpClient\Chunk\ErrorChunk;
+use Symfony\Component\HttpClient\Exception\TransportException;
+use Symfony\Component\HttpClient\PluggableHttpClient;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * Provides a single extension point to process a response's content stream.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class PluggableResponse implements ResponseInterface
+{
+    use CommonResponseTrait;
+
+    private $client;
+    private $response;
+    private $info = [];
+    private $chunkFactory;
+
+    public function __construct(HttpClientInterface $client, string $method, string $url, array $options, callable $chunkFactory)
+    {
+        $this->client = $client;
+        $this->shouldBuffer = $options['buffer'] ?? true;
+        $this->response = $client->request($method, $url, ['buffer' => false] + $options);
+        $this->chunkFactory = $chunkFactory;
+        $this->initializer = static function (self $response) {
+            return null !== $response->shouldBuffer;
+        };
+    }
+
+    public function getStatusCode(): int
+    {
+        if ($this->initializer) {
+            self::initialize($this);
+        }
+
+        return $this->response->getStatusCode();
+    }
+
+    public function getHeaders(bool $throw = true): array
+    {
+        if ($this->initializer) {
+            self::initialize($this);
+        }
+
+        $headers = $this->response->getHeaders(false);
+
+        if ($throw) {
+            $this->checkStatusCode($this->getInfo('http_code'));
+        }
+
+        return $headers;
+    }
+
+    public function getInfo(string $type = null)
+    {
+        if (null !== $type) {
+            return $this->info[$type] ?? $this->response->getInfo($type);
+        }
+
+        return $this->info + $this->response->getInfo();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toStream(bool $throw = true)
+    {
+        if ($throw) {
+            // Ensure headers arrived
+            $this->getHeaders(true);
+        }
+
+        $handle = function () {
+            $h = StreamWrapper::createResource($this->response);
```

Comment:
```suggestion
            $stream = StreamWrapper::createResource($this->response);
```

Reply:
I think `$h` is a common enough name for a stream handle, and ppl can follow one variable that spans two lines :)